### PR TITLE
Hierarchical tests in extendeddatacrossword_test and datasquare_test

### LINKED ISSUE #51: Table-driven tests
Ref https://github.com/lazyledger/rsmt2d/pull/50#discussion_r615941444

### PR DESCRIPTION
**Pull request by distractedm1nd**
Adds subtests to `extendeddatacrossword_test.go` and `datasquare_test.go` - would this already be enough to close #51 ?

### CODE REVIEW

File: extendeddatacrossword_test.go
Comment (distractedm1nd) [inline extendeddatacrossword_test.go:108]: Should an issue be created out of this TODO?

Comment (adlerjohn) [reply extendeddatacrossword_test.go:108]: Hmmm. It's not really a TODO as much as it is a note. Constructing the Merkle proofs would only be possible for a subset of non-`nil` shares, and verifying them would be a lot of extra work that isn't directly related to rsmt2d.

File: datasquare_test.go
Comment (evan-forbes) [inline datasquare_test.go:39]: my interpretation of table driven tests would be to use a format similar to https://github.com/celestiaorg/rsmt2d/blob/d771ce746cea1dad73a475a4b2d51a311476b874/rsmt2d_test.go#L15-L27

I think this would allow us to avoid a lot of code repetition and hopefully increase readability. If all of the sub-tests in a given test cannot fit in that format, then perhaps those sub tests belong in their own test.

**Review by evan-forbes — COMMENTED** [review]
left one comment that I think could be applied to most of the modifications here

thanks for digging into this @distractedm1nd !

File: datasquare_test.go
Comment (distractedm1nd) [reply datasquare_test.go:39]: Yeah that makes a lot more sense - thanks! Will see what I can do there

**Review by adlerjohn — APPROVED** [review]
Looks fine to me, will defer to @liamsi for final review.

**Comment by distractedm1nd** [comment]
Just needs another review

**Review by adlerjohn — COMMENTED** [review]
merge conflict

**Review by liamsi — APPROVED** [review]
Merge conflict needs to be resolved. Otherwise, LGTM!

**Review by liamsi — APPROVED** [review]
👏🏼

**Pull request merged**
